components/feed/SaveFeedItemButton.tsx

```diff
@@ -0,0 +1,92 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import { saveFeedItem, unsaveFeedItem } from "@/lib/savedFeedItems";
+
+const SaveFeedItemButton = ({
+  userId,
+  postId,
+  savedItemIds,
+}: {
+  userId: string;
+  postId: string;
+  savedItemIds?: string[];
+}) => {
+  const saved = savedItemIds?.includes(postId.trim());
+
+  const [localSavedState, setLocalSavedState] = useState(saved);
+
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+
+  // Create inline loading UI
+  const isMutating = isFetching || isPending;
+
+  const handleClick = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+    if (localSavedState) {
+      setLocalSavedState(!localSavedState);
+      await unsaveFeedItem(userId, postId);
+    } else {
+      setLocalSavedState(!localSavedState);
+      await saveFeedItem(userId, postId);
+    }
+    setIsFetching(false);
+    startTransition(() => {
+      // Refresh the current route and fetch new data from the server without
+      // losing client-side browser or React state.
+      router.refresh();
+    });
+  };
+
+  if (!localSavedState) {
+    return (
+      <div>
```

Comment:
This should be a button, onClick should be on it. Same with your other PR — if you can merge the two return statements and just conditionally change the SVG fill, pls do it.